src/gui/gui2_textentry.cpp

```diff
@@ -338,6 +338,10 @@ void GuiTextEntry::onTextInput(sp::TextInputEvent e)
 
 void GuiTextEntry::onFocusGained()
 {
+    if (select_on_focus) {
+		selection_end = 0;
```

Comment:
Funny thing is, I set my editor to always spaces a while ago to prevent that - and and changed it back while doing some experiments in the code lately, as the diff showed inconsistencies :-). But it seems that spaces are way more common in the code.